### PR TITLE
create GET /api/articles (queries)

### PR DESCRIPTION
**Pull request by Kayphaz007**

### CODE REVIEW

File: models/articles.models.js [hunk 1]
```diff
@@ -10,24 +10,47 @@ function selectAllUsers() {
...
-        return Promise.reject({
-          status: 404,
-          msg: "No articles found",
-        });
-      }
-      return rows.map((row) => {
-        row.votes = +row.votes;
-        row.comment_count = +row.comment_count;
-        return row;
+function selectAllArticles(topic, sort_by = "created_at", order = "desc") {
```
Comment (johnhutcheon) [inline models/articles.models.js:13]: Nice work having default values! 🙌

File: __tests__/app.test.js [hunk 2]
```diff
@@ -72,79 +72,224 @@ describe("200: /api", () => {
...
+        .then(({ body }) => {
+          const { articles } = body;
+          articles.forEach((article) => {
+            expect(article.topic).toBe("cats");
+          });
+        });
+    });
+    test("400 should return error for invalid query", () => {
+      return request(app).get("/api/articles?sort_by=cats").expect(400);
+    });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:290]: Can we just have a few more tests please? Think about what we'd expect for the following scenarios...

-  Invalid order query, e.g. ?order=bananas
- Valid topic query, but has no articles e.g. ?topic=paper. 
-  Non-existent topic query, e.g. ?topic=bananas.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -72,79 +72,224 @@ describe("200: /api", () => {
...
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles).toBeSorted({
+            key: "created_at",
+            ascending: true,
+          });
         });
+    });
+    describe("suite of test for sort_by", () => {
```
Comment (johnhutcheon) [inline __tests__/app.test.js:190]: Whilst i admire how thorough these tests are, we only really need to do one test to show sort_by works! Having a test to sort by each property is a bit overkill, as we're just displaying the same behaviour repeatedly 🙂

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Nice work so far! Could you just see my comments please and make the amendments to your testing? Thanks! 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -72,79 +72,233 @@ describe("200: /api", () => {
...
+    });
+    test("400 should return error for invalid sort query", () => {
+      return request(app).get("/api/articles?sort_by=cats").expect(400);
+    });
+    test("400 should return error for invalid order query", () => {
+      return request(app).get("/api/articles?order=banana").expect(400);
+    });
+    test("404 should return error for valid topic but has no articles", () => {
+      return request(app).get("/api/articles?topic=paper").expect(404);
+    });
```
Comment (yalicee) [inline __tests__/app.test.js]: this test should a 200

File: __tests__/app.test.js [hunk 5]
```diff
@@ -72,79 +72,233 @@ describe("200: /api", () => {
...
+    test("400 should return error for non-existent topic Query", () => {
+      return request(app).get("/api/articles?topic=banana").expect(404);
+    });
```
Comment (yalicee) [inline __tests__/app.test.js]: test description should say 404

**Review by yalicee — CHANGES_REQUESTED** [review]
John mentioned you have extra tests please remove some. Also make the changes I have requested in the comments

**Pull request merged**